### PR TITLE
NAS-104792 / 12.0 / Nas 104792

### PR DESCRIPTION
**Pull request by lola27**

### CODE REVIEW

**Comment by lola27** [comment]
@freenas/docs-team The are some fields that are specified in a different PURPOSE, so once the PURPOSE value changed, those fields will apply the specified value and will be greyed-out. Could you guys provide the tooltip for the purpose field? thanks.

File: src/app/helptext/sharing/smb/smb.ts [hunk 1]
```diff
@@ -104,6 +101,29 @@ export const helptext_sharing_smb = {
...
+    tooltip_streams: T('Enable support for Alternate Data Streams. Disabling support for streams will\
+ cause MacOS to write streams to files on the filesystem.'),
+
+    placeholder_durablehandle: T('Enable SMB2/3 durable handles'),
+    tooltip_durablehandle: T('Enable support for SMB2/3 durable handles. This parameter disables support\
+ for POSIX byte-range locks in Samba and may be unsuitable if multi-protocol (or local) access to the\
+ files is anticipated.'),
+
+    placeholder_fsrvp: T('Enable FSRVP'),
+    tooltip_fsrvp: T('Enable support for FSRVP. This protocol allows RPC clients to manage snapshots of the\
```
Comment (Mrt134) [inline src/app/helptext/sharing/smb/smb.ts]: Would it be helpful to users to spell out the acronym and make it a link to [Microsoft's documentation](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fsrvp/dae107ec-8198-4778-a950-faa7edad125b)?
```
    tooltip_fsrvp: T('Enable support for the File Server Remote VSS Protocol (FSVRP). This protocol allows \
     RPC clients to manage snapshots of the ...
```

File: src/app/helptext/sharing/smb/smb.ts [hunk 2]
```diff
@@ -104,6 +101,29 @@ export const helptext_sharing_smb = {
...
+ specified SMB share. Share path must be a dataset mountpoint. Snapshots will have a prefix "fss-" followed\
+ by the timestamp for when the snapshot was taken. RPC users will not be able to delete snapshots that don\'t\
+ have this prefix.'),
+
+    placeholder_path_suffix: T('Path Suffix'),
+    tooltip_path_suffix: T('Append a suffix (possibly containing a macro) to the share\'s connect path.\
+ This can be used to provide unique shares on a per-user, per-computer, or per-IP address basis. See\
+ <i>smb.conf</i> for list of supported macros. The connectpath _must_ be pre-staged before client connection.'),
```
Comment (Mrt134) [inline src/app/helptext/sharing/smb/smb.ts]: Propose adding a link to the relevant manual page for smb.conf:

```
<a href="https://www.freebsd.org/cgi/man.cgi?query=smb.conf" target="_blank">smb.conf(5)</a> \
 for a list of supported macros
```

File: src/app/helptext/sharing/smb/smb.ts [hunk 3]
```diff
@@ -31,6 +32,9 @@ export const helptext_sharing_smb = {
  allows for a personal, dynamic share. Only one share can be used \
  as the home share.'),
 
+    placeholder_purpose: T('Purpose'),
+    tooltip_purpose: T('Select preset configuration for share'),
```
Comment (Mrt134) [inline src/app/helptext/sharing/smb/smb.ts]:
```suggestion
    tooltip_purpose: T('Select a preset configuration for the share. Options are enabled or disabled \
     depending on the selected share purpose.'),
```

**Pull request merged**